pkg/chunk/series_store.go

```diff
@@ -440,16 +445,23 @@ func (c *seriesStore) PutOne(ctx context.Context, from, through model.Time, chun
 			return err
```

Comment:
What about this branch? If storage supports writing chunk and index at the same time, both are written, even if chunk is already stored.

(GitHub won't let me comment on line 438, but that's where it's supposed to be)

Reply:
While this change is not relevant for dynamodb which is the only store which implements `PutChunksAndIndex` with the added condition of using dynamodb for storing both index and chunks, I have changed the code to not break if there are any changes in future.